includes/Core/Authentication/Clients/Google_Site_Kit_Client.php

```diff
@@ -201,6 +203,48 @@ public function fetchAccessTokenWithRefreshToken( $refresh_token = null ) {
 		return $token_response;
 	}
 
+	/**
+	 * Executes deferred HTTP requests.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param RequestInterface $request Request object to execute.
+	 * @param string           $expected_class Expected class to return.
+	 * @return object An object of the type of the expected class or Psr\Http\Message\ResponseInterface.
+	 */
+	public function execute( RequestInterface $request, $expected_class = null ) {
+		$request = $request->withHeader( 'X-Goog-Quota-User', self::getQuotaUser() );
+
+		return parent::execute( $request, $expected_class );
+	}
+
+	/**
+	 * Returns a string that uniquely identifies a user of the application.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string Unique user identifier.
+	 */
+	public static function getQuotaUser() {
+		$url        = wp_parse_url( get_home_url() );
+		$quota_user = "{$url['scheme']}://";
+
+		$user        = wp_get_current_user();
+		$quota_user .= $user instanceof WP_User && $user->ID
+			? "{$user->user_login}@{$url['host']}"
+			: $url['host'];
+
+		if ( ! empty( $url['port'] ) ) {
+			$quota_user .= ":{$url['port']}";
+		}
```

Comment:
I think this part probably isn't necessary as Site Kit doesn't support non-default ports.

Reply:
Updated.